src/screen/show_refs.rs

```diff
@@ -18,57 +22,122 @@ pub(crate) fn create(config: Rc<Config>, repo: Rc<Repository>, size: Rect) -> Re
-                            } else {
-                                "  "
-                            },
-                        );
-
-                        let upstream_name = if let Ok(upstream) = branch.upstream() {
-                            if let Ok(Some(name)) = upstream.name() {
-                                Span::styled(name.to_string(), &style.remote)
-                            } else {
-                                Span::raw("")
-                            }
-                        } else {
-                            Span::raw("")
-                        };
-
-                        Item {
-                            id: name.clone().content,
-                            display: Line::from(vec![
-                                prefix,
-                                name.clone(),
-                                Span::raw("   "),
-                                upstream_name,
-                            ]),
-                            depth: 1,
-                            target_data: Some(TargetData::Branch(name.content.into())),
-                            ..Default::default()
-                        }
-                    }),
-            )
+            .chain(create_references_section(
+                &repo,
+                Reference::is_branch,
+                &style.branch,
+            )?)
+            .chain(create_remotes_sections(
+                &repo,
+                &style.section_header,
+                &style.remote,
+            )?)
+            .chain(iter::once(Item {
+                id: "tags".into(),
+                display: Line::styled("Tags".to_string(), &style.section_header),
+                section: true,
+                depth: 0,
+                ..Default::default()
+            }))
+            .chain(create_references_section(
+                &repo,
+                Reference::is_tag,
+                &style.tag,
+            )?)
             .collect())
         }),
     )
 }
+
+fn create_remotes_sections<'a>(
+    repo: &'a Repository,
+    header_style: &'a StyleConfigEntry,
+    item_style: &'a StyleConfigEntry,
+) -> Res<impl Iterator<Item = Item> + 'a> {
+    let all_remotes = create_references_section(repo, Reference::is_remote, item_style)?;
+    let mut remotes = HashMap::new();
+    for remote in all_remotes {
+        let name = match remote.id.split_once('/') {
+            None => remote.id.as_ref(),
+            Some((name, _)) => name,
+        };
+
+        match remotes.entry(name.to_string()) {
+            Entry::Vacant(entry) => {
+                entry.insert(vec![remote]);
+            }
+            Entry::Occupied(mut entry) => {
+                entry.get_mut().push(remote);
+            }
+        }
+    }
+
+    Ok(remotes.into_iter().flat_map(move |(name, items)| {
+        let header = capitalize(&name);
+        iter::once(Item {
+            id: name.into(),
+            display: Line::styled(header, header_style),
```

Comment:
I'd just copy Magit and have these headers be e.g. `Remote origin`, makes it clear that it is a remote.
It is simpler to not capitalize the origin names I think.
Tested it, and it is actually possible to have `Origin` and `origin` simultaneously in git.

Reply:
Done